native/cocos/platform/interfaces/modules/Device.cpp

```diff
@@ -30,9 +30,32 @@
 #include "platform/interfaces/modules/INetwork.h"
 #include "platform/interfaces/modules/IScreen.h"
 #include "platform/interfaces/modules/IVibrator.h"
+#include "platform/interfaces/modules/ISystemWindow.h"
+#include "platform/interfaces/modules/ISystemWindowManager.h"
 
 namespace cc {
 
+uint32_t Device::getInnerWidth() {
+    ISystemWindow* systemWindow = CC_GET_MAIN_SYSTEM_WINDOW();
+    CC_ASSERT_NOT_NULL(systemWindow);
+    auto viewSize = systemWindow->getViewSize();
+    return static_cast<uint32_t>(viewSize.width);
+}
+
+uint32_t Device::getInnerHeight() {
+    ISystemWindow* systemWindow = CC_GET_MAIN_SYSTEM_WINDOW();
```

Comment:
ISystemWindow was bound, could we get ISystemWindow in JS directly?

Reply:
Only the interface is exported, not the instance? This instance cannot be created directly.

Reply:
How about using `root.mainWindow.width/height`?

Reply:
If we can use this interface, can't window.innerWidth be deleted?

Reply:
Should not delete `window.innerWidth` which is for keeping compatibility with `web`.
So where do you use `Device::getInnerHeight` after you exported `Device::getInnerHeight` to JS?